shared/protocol/src/main/java/io/pravega/shared/metrics/ClientMetricKeys.java

```diff
@@ -0,0 +1,56 @@
+/**
+ * Copyright (c) Dell Inc., or its subsidiaries. All Rights Reserved.
+ *
+ * Licensed under the Apache License, Version 2.0 (the "License");
+ * you may not use this file except in compliance with the License.
+ * You may obtain a copy of the License at
+ *
+ *     http://www.apache.org/licenses/LICENSE-2.0
+ */
+package io.pravega.shared.metrics;
+
+import com.google.common.annotations.VisibleForTesting;
+import com.google.common.base.Preconditions;
+import com.google.common.base.Strings;
+import lombok.Getter;
+
+/**
+ * The various metrics that can be generated by the client.
+ */
+public enum ClientMetricKeys {
+
+    CLIENT_APPEND_LATENCY("client.segment.append_latency_ms"),
+    CLIENT_APPEND_BLOCK_SIZE("client.segment.append_block_size");
```

Comment:
To have a complete understanding of the performance of the client, we are missing one metric: the `append_response_latency_ms`. That is, `append_latency_ms` gives use the time to write the event to the server, but as I posted in some plots, this differs from the total latency an application sees waiting for the write ack to complete. I think we can easily add this metric in `FlowHandler.channelRead` when handling `WireCommands.DataAppended` events, right?

Reply:
Thanks @RaulGracia . Looked into enabling metric `append_response_latency_ms`, time taken to receive an ack for a given event. Do not see a way to enable it without an additional penalty on the client-side. We could track this in a different issue.
Based on your comment did the following changes
- Ensured the segment writer id is part of the metric, since there can be multiple writers to the same segment.
- Added a new metric `CLIENT_OUTSTANDING_APPEND_COUNT`. This metric represents the number of appends for which we have not yet received acks from the segment store.